agent/agent.go

```diff
@@ -91,35 +91,36 @@ func (a *Agent) Run(payload *queue.Work, cancel <-chan bool) error {
 }
 
 func (a *Agent) prep(w *queue.Work) (*yaml.Config, error) {
-
 	envs := toEnv(w)
 	w.Yaml = expander.ExpandString(w.Yaml, envs)
 
-	// inject the netrc file into the clone plugin if the repository is
-	// private and requires authentication.
+	// append all secrets. they will be checked by the ImageSecrets transform.
 	var secrets []*model.Secret
-	if w.Verified {
-		secrets = append(secrets, w.Secrets...)
-	}
+	secrets = append(secrets, w.Secrets...)
```

Comment:
if we check skipverify here, we probably don't need to modify the transform function

```
	var secrets []*model.Secret

	for _, secret := range w.Secrets {
		if w.Verified || secret.SkipVerify {
				secrets = append(secrets, secret)
		}
	}
	
```